src/parsers/containers/isobmff/utils.ts

```diff
@@ -190,60 +168,59 @@ function getTrackFragmentDecodeTime(buffer : Uint8Array) : number {
   if (traf === null) {
     return -1;
   }
-
-  const index = findBox(traf, 0x74666474 /* tfdt */);
-  if (index === -1) {
+  const tfdt = getBoxContent(traf, 0x74666474 /* tfdt */);
+  if (tfdt === null) {
     return -1;
   }
-
-  let pos = index + /* size */4 + /* name */4;
-  const version = traf[pos]; pos += 4;
-  if (version > 1) {
-    return -1;
-  }
-
-  return version !== 0 ? be8toi(traf, pos) :
-                         be4toi(traf, pos);
+  const version = tfdt[0];
+  return version === 1 ? be8toi(tfdt, 4) :
+         version === 0 ? be4toi(tfdt, 4) :
+                         -1;
```

Comment:
Maybe we should return undefined in case we can't return a tfdt, nope ?

Reply:
Globally, in the utils file, why do we return `-1 `?

Reply:
Done